### DISCUSSION TITLE
Question about the validator of `Transition`

### DISCUSSION THREAD

**Discussion by cymqqqq** [RGB-WG/rgb#232] [Q&A] [answered]
I found a `validator` in both the `TransitionSchema` and the `Transition`, and then the `validator` in the `TransitionSchema` now is defined as `Option<LibSite>`. But the `validator` in the `Transition` is still the `ReservedBytes<1, 0>`, so can we also define it as `Option<LibSite>`, then we can get the scripts from the schema and implement `// TODO: Validate against schema` by rgbisa?

**Comment by cymqqqq** [comment]
The same for `Genesis`, and `Extension`

**Comment by dr-orlovsky** [comment, marked as answer]
This is reserved for the future RGB versions, since requires a lot of formal verification on the ramifications of custom transition-based validation logic and will not be worked on before v0.12
